### DISCUSSION TITLE
generating only the necessary part of the schema

### DISCUSSION THREAD

**Discussion by killjoy2013** [dotansimha/graphql-code-generator#9099] [General] [answered]
Hi,
Using Apollo Federation 2. Using graphql-request for syncronous communication between subgraphs. 
In one of my subgraph (food) I need to run `city` query to get city info which is handled by another subgraph (country). food microservice knows nothing about country microservice. It just runs the city query against federated schema (supergraph). I'm using codegen to generate graphql types using supergraph;

codegen.ts

```ts
import { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  schema: '../router/supergraph-schema.graphql',
  documents: ['./src/gql_definitions/queries.ts'],
  ignoreNoDocuments: true,
  generates: {
    './src/gql/': {
      preset: 'client',
      plugins: [],
    },
  },
};

export default config;
```

My query is;

```ts
import { graphql } from '../gql/gql';

export const CITY = graphql(`
  query City($cityId: Int!) {
    city(id: $cityId) {
      id
      name
      population
    }
  }
`);
```

finally, below files created successfully;

```
./gql/fragment-masking.ts
./gql/gql.ts
./gql/graphql.ts
./gql/index.ts
```
However, all the super schema entities are generated in `graphql.ts`. This causes unnecessary builds of my food microservice just because of some changes which are not related to my food service, it's using only city query. In a monorepo environment, I'm trying to save the microservices from unnecessary builds. 

Is it possible to generate only the needed part of the schema? Here we use city query & need only related types to this query.

I love graphql codegenerator, thank you all for it 😄

**Comment by n1ru4l** [comment, marked as answer]
This is already on the roadmap. https://github.com/dotansimha/graphql-code-generator/issues/8296

<img width="324" alt="image" src="https://user-images.githubusercontent.com/14338007/223058510-4cfa38ce-6b21-4bd7-b4c0-a2a7fe429425.png">

**Reply by killjoy2013** [reply]
That made me really happy 👏 

when can we test?

<img width="891" alt="Screen Shot 2023-03-06 at 11 58 47" src="https://user-images.githubusercontent.com/6085821/223063764-2c6a80b9-57eb-4ea5-bf37-703836edaf0f.png">
